src/web-components/KSLAddButtonElement.ts

```diff
@@ -236,11 +244,21 @@ export class KSLAddButtonElement extends KSLPositionedElement {
       );
 
       const { permissions } = response;
+      const isUserMissingPermissions =
+        !permissions ||
+        permissions.get(AddButtonPermission.ViewParent) !== AddButtonPermissionCheckResult.Ok ||
+        permissions.get(AddButtonPermission.Edit) !== AddButtonPermissionCheckResult.Ok;
 
-      if (!permissions || permissions.get(AddButtonPermission.ViewParent) !== AddButtonPermissionCheckResult.Ok) {
+      if (isUserMissingPermissions) {
         this.buttonRef.loading = false;
         this.buttonRef.disabled = true;
         this.buttonRef.tooltipMessage = 'You are not allowed to add content here';
+      } else if (
+        permissions.get(AddButtonPermission.CreateNew) === AddButtonPermissionCheckResult.RteWithForbiddenComponents
+      ) {
+        this.buttonRef.loading = false;
+        this.buttonRef.disabled = true;
+        this.buttonRef.tooltipMessage = "Components and items can't be added here";
```

Comment:
nitpick: this if block looks the same as the previous one, except for the tooltip message, can we maybe combine them into one block?

Reply:
Nice catch, thank you. I got rid of the unnecessary if-else.